pkgs/by-name/gl/glide-media-player/package.nix

```diff
@@ -0,0 +1,86 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, meson
+, ninja
+, rustc
+, cargo
+, wrapGAppsHook4
+, python3
+, libadwaita
+, graphene
+, gst_all_1
+, glib-networking
+, darwin
+, libsoup_3
+}:
+
+stdenv.mkDerivation rec {
+  pname = "glide-media-player";
+  version = "0.6.1";
+
+  src = fetchFromGitHub {
+    owner = "philn";
+    repo = "glide";
+    rev = version;
+    hash = "sha256-dIXuWaoTeyVBhzr6VWxYBsn+CnUYG/KzhzNJtLLdRuI=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-azvxW40fuKuF/N0qwzofFk1bZiNxyTN6YBFU5qHQkCA=";
+  };
+
+  postPatch = ''
+    substituteInPlace scripts/meson_post_install.py \
+      --replace "gtk-update-icon-cache" "gtk4-update-icon-cache"
+    patchShebangs --build scripts/meson_post_install.py
+  '' + lib.optionalString stdenv.isDarwin ''
+    sed -i "/wayland,x11egl,x11glx/d" meson.build
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+    meson
+    ninja
+    rustPlatform.cargoSetupHook
+    rustc
+    cargo
+    wrapGAppsHook4
+    python3
+  ];
+
+  buildInputs = [
+    libadwaita
+    graphene
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-bad
+    gst_all_1.gst-plugins-good
+    glib-networking
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk_11_0.frameworks.IOKit
+  ];
+
+  # FIXME: gst-plugins-good missing libsoup breaks streaming
+  # (https://github.com/nixos/nixpkgs/issues/271960)
+  preFixup = ''
+    gappsWrapperArgs+=(--prefix LD_LIBRARY_PATH : "${
+      lib.makeLibraryPath [ libsoup_3 ]
+    }")
+  '';
```

Comment:
Can you rather change this in gst-plugins-good and target staging? Using LD_LIBRARY_PATH is generally a bad solution and should be avoided as much as possible

Reply:
this seems like an unreasonable request -- there is an issue to fix this https://github.com/NixOS/nixpkgs/issues/271960 but no one has yet come up with a reasonable solution. the base issue is that dlopen searches the rpath of the calling shared object. given gstreamer uses g_module_open which is a glib function, the issue is that dlopen is called by glib/libgmodule.so, so to fix this you are requesting that the glib code be updated, which seems unreasonable.

Reply:
The issue is a bit misleading. The fix is not as simple as just adding libsoup to gst-plugins-good which the issues especially from the first comments suggests.

This means that the comment and the issue title should be updated to better reflect this otherwise the next person is thinking the same thing as me.

Reply:
this was also already explained previously in this PR, https://github.com/NixOS/nixpkgs/pull/275391#discussion_r1435222327 and you made a similar comment in another PR which i also explained it to you https://github.com/NixOS/nixpkgs/pull/271975#discussion_r1418055980

regardless, getting gstreamer to work out of the box would be nice.

Reply:
> Can you rather change this in gst-plugins-good and target staging? Using LD_LIBRARY_PATH is generally a bad solution and should be avoided as much as possible

I have no idea... I don't really use this kind of feature and I'm not very familiar with gstreamer.

Reply:
I have a fix that should eliminate the workaround. https://github.com/NixOS/nixpkgs/pull/276408 if it gets merged I'll come back around and remove the workaround in this change.